src/ray/core_worker/actor_manager.h

```diff
@@ -22,6 +22,46 @@
 
 namespace ray {
 
+class ActorCreatorInterface {
+ public:
+  virtual ~ActorCreatorInterface() = default;
+  /// Register actor to GCS asynchronously.
+  ///
+  /// \param task_spec The specification for the actor creation task.
+  /// \param callback Callback that will be called after the actor info is written to GCS.
+  /// \return Status
+  virtual Status AsyncRegisterActor(const TaskSpecification &task_spec,
+                                    const gcs::StatusCallback &callback) = 0;
+
+  /// Report actor dependencies resolved to GCS asynchronously.
+  ///
+  /// \param task_spec The specification for the actor creation task.
+  /// \param callback Callback that will be called after the actor info is written to GCS.
+  /// \return Status
+  virtual Status AsyncReportActorDependenciesResolved(
+      const TaskSpecification &task_spec, const gcs::StatusCallback &callback) = 0;
+};
+
+class DefaultActorCreator : public ActorCreatorInterface {
+ public:
+  explicit DefaultActorCreator(std::shared_ptr<gcs::GcsClient> gcs_client)
+      : gcs_client_(std::move(gcs_client)) {}
+
+  Status AsyncRegisterActor(const TaskSpecification &task_spec,
+                            const gcs::StatusCallback &callback) override {
+    return gcs_client_->Actors().AsyncRegisterActor(task_spec, callback);
+  }
+
+  Status AsyncReportActorDependenciesResolved(
```

Comment:
I think `AsyncCreateActor` sounds better? The function name sounds a little misleading to me because the response will come back after the actor is created.

Reply:
> I think `AsyncCreateActor` sounds better? The function name sounds a little misleading to me because the response will come back after the actor is created.

I think naming should be consistent with behavior. What do you think? @raulchen

Reply:
@rkooo567 The fact that the reponse will come back after the actor is created is just an implementation detail. We could change it in the future. 
While I think it makes more sense to name this method after its purpose, which is to report dependencies resolved. 
Also, in the future, we can optimize the process by skipping `AsyncReportActorDependenciesResolved`, if the dependencies are already ready in `RegisterActor` .

Reply:
I think the current interface rather exposes implementation details. This RPC initiates the actor creation process and reply comes after the actor is actually created (Let me know if this is wrong). `AsyncReportActorDependenciesResolved` is indeed an implementation details (It is the first step to create an actor).

Reply:
@rkooo567 Let me put it this way. Creating an actor includes multiple steps:
1. Client registers the actor to the GCS.
2. Client reports actor dependencies ready.
3. GCS schedules the actor.

So the method should be named after these steps. If we name this method `CreateActor`, it could be confusing because the first step (register) is also part of the actor creation steps.

Also, in the future, we can add an optimization to skip step 2 if depdencies are already ready at step 1. At that time, this name would become even more weird. 

Please let me know whether this makes sense.